### PR TITLE
Starting JBossEAP using systemd | Opening additional ports 

### PR DESCRIPTION
**Pull request by Pbasnal**

### CODE REVIEW

**Comment by mnovak1** [comment]
@Pbasnal hi, with @istraka we'll take a look at this MR and provide review.

File: eap73-rhel8-byos-multivm/scripts/jbosseap-setup-redhat.sh [hunk 1]
```diff
@@ -126,42 +133,30 @@ echo "Copy the standalone-azure-ha.xml from EAP_HOME/doc/wildfly/examples/config
...
+echo -e "\t private:inet-address"       | adddate >> /var/log/jbosseap.install.log
+echo -e "\t webservices:wsdl-host"      | adddate >> /var/log/jbosseap.install.log
+
+$EAP_HOME/wildfly/bin/jboss-cli.sh --echo-command \
+'embed-server --std-out=echo  --server-config=standalone-azure-ha.xml',\
+'/subsystem=jgroups/channel=ee:write-attribute(name="stack", value="tcp")',\
+'/interface=management:write-attribute(name=inet-address, value="${jboss.bind.address.management:0.0.0.0}")',\
+'/interface=public:write-attribute(name=inet-address, value="${jboss.bind.address:0.0.0.0}")',\
+'/interface=private:write-attribute(name=inet-address, value="${jboss.bind.address.private:0.0.0.0}")',\
+'/subsystem=webservices:write-attribute(name=wsdl-host, value="${jboss.bind.address:0.0.0.0}")' | adddate >> /var/log/jbosseap.install.log 2>&1
```
Comment (mnovak1) [inline eap73-rhel8-byos-multivm/scripts/jbosseap-setup-redhat.sh:149]: On the other hand, this one should be set to Azure VM public IP if this should work outside Azure cloud. I think we should not set this one and leave default there.

File: eap73-rhel8-byos-vmss/scripts/jbosseap-setup-redhat.sh [hunk 2]
```diff
@@ -53,14 +60,14 @@ echo "RHSM_USER: " ${RHSM_USER} | adddate >> /var/log/jbosseap.install.log
...
-echo "firewall-cmd --zone=public --add-port=9990/tcp  --permanent"  | adddate >> /var/log/jbosseap.install.log
-sudo firewall-cmd  --zone=public --add-port=9990/tcp  --permanent   | adddate >> /var/log/jbosseap.install.log 2>&1
-echo "firewall-cmd --zone=public --add-port=45700/tcp --permanent" | adddate >> /var/log/jbosseap.install.log
-sudo firewall-cmd  --zone=public --add-port=45700/tcp --permanent  | adddate >> /var/log/jbosseap.install.log 2>&1
-echo "firewall-cmd --zone=public --add-port=7600/tcp  --permanent"  | adddate >> /var/log/jbosseap.install.log
-sudo firewall-cmd  --zone=public --add-port=7600/tcp  --permanent   | adddate >> /var/log/jbosseap.install.log 2>&1
+openport 9999
+openport 8443
+openport 8009
+openport 5445
```
Comment (mnovak1) [inline eap73-rhel8-byos-vmss/scripts/jbosseap-setup-redhat.sh:66]: can be removed as for multivm template

File: eap73-rhel8-byos-vmss/scripts/jbosseap-setup-redhat.sh [hunk 3]
```diff
@@ -117,41 +124,30 @@ echo "Copy the standalone-azure-ha.xml from EAP_HOME/doc/wildfly/examples/config
...
+echo -e "\t private:inet-address"       | adddate >> /var/log/jbosseap.install.log
+echo -e "\t webservices:wsdl-host"      | adddate >> /var/log/jbosseap.install.log
+
+$EAP_HOME/wildfly/bin/jboss-cli.sh --echo-command \
+'embed-server --std-out=echo  --server-config=standalone-azure-ha.xml',\
+'/subsystem=jgroups/channel=ee:write-attribute(name="stack", value="tcp")',\
+'/interface=management:write-attribute(name=inet-address, value="${jboss.bind.address.management:0.0.0.0}")',\
+'/interface=public:write-attribute(name=inet-address, value="${jboss.bind.address:0.0.0.0}")',\
+'/interface=private:write-attribute(name=inet-address, value="${jboss.bind.address.private:0.0.0.0}")',\
+'/subsystem=webservices:write-attribute(name=wsdl-host, value="${jboss.bind.address:0.0.0.0}")' | adddate >> /var/log/jbosseap.install.log 2>&1
```
Comment (mnovak1) [inline eap73-rhel8-byos-vmss/scripts/jbosseap-setup-redhat.sh:140]: same as for multivm template

**Review by istraka — COMMENTED** [review]
Suggested changes need to be also applied on PAYG templates.

File: eap73-rhel8-byos-multivm/scripts/jbosseap-setup-redhat.sh [hunk 4]
```diff
@@ -126,42 +133,30 @@ echo "Copy the standalone-azure-ha.xml from EAP_HOME/doc/wildfly/examples/config
...
+echo -e "\t public:inet-address"        | adddate >> /var/log/jbosseap.install.log
+echo -e "\t private:inet-address"       | adddate >> /var/log/jbosseap.install.log
+echo -e "\t webservices:wsdl-host"      | adddate >> /var/log/jbosseap.install.log
+
+$EAP_HOME/wildfly/bin/jboss-cli.sh --echo-command \
+'embed-server --std-out=echo  --server-config=standalone-azure-ha.xml',\
+'/subsystem=jgroups/channel=ee:write-attribute(name="stack", value="tcp")',\
+'/interface=management:write-attribute(name=inet-address, value="${jboss.bind.address.management:0.0.0.0}")',\
+'/interface=public:write-attribute(name=inet-address, value="${jboss.bind.address:0.0.0.0}")',\
+'/interface=private:write-attribute(name=inet-address, value="${jboss.bind.address.private:0.0.0.0}")',\
```
[the remark this answers is not in the document]
Comment (istraka) [reply eap73-rhel8-byos-multivm/scripts/jbosseap-setup-redhat.sh:148]: `$(hostname -I)` won't be resolved in `/etc/opt/rh/eap7/wildfly/eap7-standalone.conf`. See [the documentation](https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.3/html-single/configuration_guide/index#configure_rpm_service_properties). Better put it into `/opt/rh/eap7/root/usr/share/wildfly/bin/standalone.conf`

File: eap73-rhel8-byos-multivm/scripts/jbosseap-setup-redhat.sh [hunk 5]
```diff
@@ -55,14 +62,14 @@ echo "Folder where script is executing ${pwd}" | adddate >> /var/log/jbosseap.in
...
-echo "firewall-cmd --zone=public --add-port=9990/tcp  --permanent"  | adddate >> /var/log/jbosseap.install.log
-sudo firewall-cmd  --zone=public --add-port=9990/tcp  --permanent   | adddate >> /var/log/jbosseap.install.log 2>&1
-echo "firewall-cmd --zone=public --add-port=45700/tcp --permanent"  | adddate >> /var/log/jbosseap.install.log
-sudo firewall-cmd  --zone=public --add-port=45700/tcp --permanent   | adddate >> /var/log/jbosseap.install.log 2>&1
-echo "firewall-cmd --zone=public --add-port=7600/tcp  --permanent"  | adddate >> /var/log/jbosseap.install.log
-sudo firewall-cmd  --zone=public --add-port=7600/tcp  --permanent   | adddate >> /var/log/jbosseap.install.log 2>&1
+openport 9999
+openport 8443
+openport 8009
+openport 5445
```
[the remark this answers is not in the document]
Comment (Pbasnal) [reply eap73-rhel8-byos-multivm/scripts/jbosseap-setup-redhat.sh]: removed

File: eap73-rhel8-byos-multivm/scripts/jbosseap-setup-redhat.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (Pbasnal) [reply eap73-rhel8-byos-multivm/scripts/jbosseap-setup-redhat.sh]: Updated the script.

File: eap73-rhel8-byos-multivm/scripts/jbosseap-setup-redhat.sh (hunk 1 above)
Comment (Pbasnal) [reply eap73-rhel8-byos-multivm/scripts/jbosseap-setup-redhat.sh]: done

File: eap73-rhel8-byos-vmss/scripts/jbosseap-setup-redhat.sh (hunk 3 above)
Comment (Pbasnal) [reply eap73-rhel8-byos-vmss/scripts/jbosseap-setup-redhat.sh:140]: changed

**Review by istraka — COMMENTED** [review]
There is an issue with `jboss.bind.address.private`. It needs to be addressed in payg and byos clustering tempaltes.

File: eap73-rhel8-byos-vmss/scripts/jbosseap-setup-redhat.sh [hunk 6]
```diff
@@ -115,43 +123,45 @@ systemctl restart sshd | adddate >> /var/log/jbosseap.install.log 2>&1
...
+echo "Setting configurations in $EAP_RPM_CONF_STANDALONE"
+echo -e "\t-> WILDFLY_SERVER_CONFIG=standalone-azure-ha.xml" | log_info 
+echo 'WILDFLY_SERVER_CONFIG=standalone-azure-ha.xml' >> $EAP_RPM_CONF_STANDALONE 2>log_err | log_info
+
+echo -e "\t-> WILDFLY_OPTS=-Djboss.bind.address.management=0.0.0.0" | log_info 
+echo 'WILDFLY_OPTS="-Djboss.bind.address.management=0.0.0.0"' >> $EAP_RPM_CONF_STANDALONE 2>log_err | log_info
+
+echo "Setting configurations in $EAP_LAUNCH_CONFIG"
+echo -e '\t-> JAVA_OPTS="$JAVA_OPTS -Djboss.bind.address.management=$(hostname -I)"' | log_info 
+echo 'JAVA_OPTS="$JAVA_OPTS -Djboss.bind.address.management=$(hostname -I)"' >> $EAP_LAUNCH_CONFIG 2>log_err | log_info
```
Comment (mnovak1) [inline eap73-rhel8-byos-vmss/scripts/jbosseap-setup-redhat.sh:149]: this should be bound to 0.0.0.0 as well.

**Pull request merged**